### ISSUE TITLE
New Rule: Consider decorating method with @staticmethod PYL-R0201

### ISSUE THREAD

**Issue by Zerotask** [astral-sh/ruff#11853]
Pycharm and Deepsource warns you, when you use an instance method which does not use `self` and therefore can be marked with `@staticmethod`.

> The method doesn't use its bound instance. Decorate this method with @staticmethod decorator, so that Python does not have to instantiate a bound method for every instance of this class thereby saving memory and computation. Read more about staticmethods [here](https://docs.python.org/3/library/functions.html#staticmethod).

It would be great to have that check in ruff as well.

Example:
```python
class Calculator:
    def add(self, a, b):
        return a + b
``` 

could be:
```python
class Calculator:
    @staticmethod
    def add(a, b):
        return a + b
```

**Comment by matteo-zanoni** [comment]
This rule has been renamed [here](https://pylint.pycqa.org/en/v3.2.3/user_guide/messages/refactor/old-no-self-use.html).
It is now **R6301** that is [already implemented](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/pylint/rules/no_self_use.rs) in ruff.

**Comment by Zerotask** [comment]
Thank you. I was blind. I actually marked the line with `# noqa: PLR6301` :D
I can confirm that it works like expected - without the noqa
